invenio_rdm_records/services/permissions.py

```diff
@@ -112,6 +112,8 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
             else_=can_read,
         )
     ]
+    can_read_deleted_files = can_read_deleted
+    can_media_read_deleted_files = can_read_deleted_files
```

Comment:
same as above

Reply:
For getting the content of the file we are also checking the permissions if it's deleted see [here](https://github.com/inveniosoftware/invenio-records-resources/pull/511/files#diff-13fd70056deb6848af5488fc2db9e690a05806d34b2b9333218e37bc60168d4bR285)